packages/relay/src/lib/services/metricService/metricService.ts

```diff
@@ -0,0 +1,354 @@
+      registers: [register],
+    });
+  }
+
+  /**
+   * Captures and records metrics for a transaction.
+   * @private
+   * @param {string} mode - The mode of the transaction (e.g., consensus mode).
+   * @param {string} type - The type of the transaction.
+   * @param {Status} status - The status of the transaction.
+   * @param {number} cost - The cost of the transaction in tinybars.
+   * @param {number} gas - The gas used by the transaction.
+   * @param {string} caller - The name of the caller executing the transaction.
+   * @param {string} interactingEntity - The entity interacting with the transaction.
+   * @returns {void}
+   */
+  private captureMetrics = (
+    mode: string,
+    type: string,
+    status: Status,
+    cost: number,
+    gas: number,
+    caller: string,
+    interactingEntity: string,
+  ): void => {
+    this.consensusNodeClientHistogramCost
+      .labels(mode, type, status.toString(), caller, interactingEntity)
+      .observe(cost);
+    this.consensusNodeClientHistogramGasFee
+      .labels(mode, type, status.toString(), caller, interactingEntity)
+      .observe(gas);
+  };
+
+  /**
+   * Retrieves transaction record metrics based on the transaction ID.
+   * Depending on the environment configuration, the metrics are fetched either from the
+   * consensus node via the SDK client or from the mirror node.
+   *
+   * @param {string} transactionId - The ID of the transaction for which metrics are being retrieved.
+   * @param {string} callerName - The name of the caller requesting the metrics.
+   * @param {string} requestId - The request ID for tracing the request flow.
+   * @param {string} txConstructorName - The name of the transaction constructor.
+   * @param {string} operatorAccountId - The account ID of the operator.
+   * @returns {Promise<ITransactionRecordMetric | undefined>} - The transaction record metrics or undefined if retrieval fails.
+   */
+  private async getTransactionRecordMetrics(
+    transactionId: string,
+    callerName: string,
+    requestId: string,
+    txConstructorName: string,
+    operatorAccountId: string,
+  ): Promise<ITransactionRecordMetric | undefined> {
+    const formattedRequestId = formatRequestIdMessage(requestId);
+    const defaultToConsensusNode = process.env.GET_RECORD_DEFAULT_TO_CONSENSUS_NODE === 'true';
+
+    // retrieve transaction metrics
+    try {
+      if (defaultToConsensusNode) {
+        return await this.sdkClient.getTransactionRecordMetrics(
+          transactionId,
+          callerName,
+          requestId,
+          txConstructorName,
+          operatorAccountId,
+        );
+      } else {
+        return await this.mirrorNodeClient.getTransactionRecordMetrics(
+          transactionId,
+          callerName,
+          requestId,
+          txConstructorName,
+          operatorAccountId,
+        );
+      }
+    } catch (error: any) {
+      this.logger.warn(error, `${formattedRequestId} ${error.message}`);
```

Comment:
Could make this more descriptive e.g Couldn't fetch transactionRecord due to error: _**the error message**_

Reply:
I like the idea. Updated!